### PR TITLE
Resolve SPM dependencies when only local packages are present

### PR DESCRIPTION
**Pull request by muukii**
### Short description 📝

To solve an issue that won't make .package.resolved file in case of using a local package but that package uses remote packages internally.

image

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

### Contributor checklist ✅

- [x] The code has been linted using run `make workspace/lint-fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistGenerator/Utils/SwiftPackageManagerInteractor.swift [hunk 1]
```diff
@@ -47,8 +47,8 @@ public class SwiftPackageManagerInteractor: SwiftPackageManagerInteracting {
         config: Config,
         graphTraverser: GraphTraversing
     ) async throws {
-        guard !config.generationOptions.disablePackageVersionLocking,
-              graphTraverser.hasRemotePackages
```
Comment (muukii) [inline Sources/TuistGenerator/Utils/SwiftPackageManagerInteractor.swift:51]: I could not see how the side effects by removing this check

**Review by danyf90 — APPROVED** [review]

File: Sources/TuistGenerator/Utils/SwiftPackageManagerInteractor.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/TuistGenerator/Utils/SwiftPackageManagerInteractor.swift:51]: won't this try to install packages even if there are none defined? I think this check was here for a reason but we can run install if there is a remote _or_ a local package

**Review by fortmarek — CHANGES_REQUESTED** [review]

File: Sources/TuistGenerator/Utils/SwiftPackageManagerInteractor.swift (hunk 1 above)
Comment (muukii) [reply Sources/TuistGenerator/Utils/SwiftPackageManagerInteractor.swift:51]: I just added check if there is package deps.
seems it's fine
https://github.com/tuist/tuist/pull/5614/commits/07237f47f3a760983f44a8aa3b4eb82c194888ac#diff-7e65e9edc17f330d07a575ad33674f3b60d1b8d190a84aec9f8d616ebfe01d86R52

Comment (muukii) [reply Sources/TuistGenerator/Utils/SwiftPackageManagerInteractor.swift:51]: OK, I'll try to add that.

File: Tests/TuistGeneratorIntegrationTests/Generator/SwiftPackageManagerInteractorTests.swift [hunk 2]
```diff
@@ -231,6 +231,58 @@ final class SwiftPackageManagerInteractorTests: TuistUnitTestCase {
...
+            plugins: [],
+            generationOptions: .test(clonedSourcePackagesDirPath: temporaryPath.appending(component: "spm")),
+            path: nil
+        )
+
+        let spmPath = temporaryPath.appending(component: "spm")
+        let target = anyTarget(dependencies: [
+            .package(product: "Example", type: .runtime),
+        ])
+        let package = Package.local(path: "/some/local/package")
```
Comment (muukii) [inline Tests/TuistGeneratorIntegrationTests/Generator/SwiftPackageManagerInteractorTests.swift:251]: using local as umbrella

**Review by fortmarek — APPROVED** [review]
Thanks again for contributing 🙌

**Comment by muukii** [comment]
@fortmarek I fixed lint as it failed.

**Comment by muukii** [comment]
@danyf90 @fortmarek Who merge this PR? or still waiting for CI passed? I'm not sure why the acceptance testing were failed. I might need you to help this.

**Comment by fortmarek** [comment]
> or still waiting for CI passed? I'm not sure why the acceptance testing were failed. I might need you to help this.

Yeah, the acceptance test [failure](https://github.com/tuist/tuist/actions/runs/7076929777/job/19275883475?pr=5614#step:9:26) seems like it could be related to your changes. Have you tried running those tests locally?

We have docs for how to run acceptance tests [here](https://docs.tuist.io/contributors/testing-strategy/#acceptance-tests). We're in the process of rewriting these to Swift, so if you want to help with that effort, you could try rewriting the failing test. The rewritten test would go [here](https://github.com/tuist/tuist/blob/main/Tests/TuistGenerateOneAcceptanceTests/GenerateOneAcceptanceTests.swift) and it would be quite similar to the other existing acceptance tests there.

**Comment by muukii** [comment]
@fortmarek I see. I will do something. btw ./fourier is no longer available, right? 
I'll see what happens in running test command in makefile.
Would you do something for testing for me?

**Comment by muukii** [comment]
ok, I got an error from here.
I'll get to the bottom of it.
image

**Comment by fortmarek** [comment]
Thanks @muukii!

Let me know if you need any help. The best way to run those acceptance tests is by selecting the `TuistGenerateOneAcceptanceTests` scheme and running the single test. That way, you should reliably see the logs from that test. You should also be able to set breakpoints that should help you debug the issue.

**Comment by danyf90** [comment]
Closing this to reduce noise, @muukii  feel free to reopen if you resume working on it 🚀

**Pull request closed without merging**